src/app/students/(table)/student-table.tsx

```diff
@@ -38,10 +40,50 @@ const DataTable = <TData, TValue>({
     columns,
     data,
 }: DataTableProps<TData, TValue>) => {
-    const [rowSelection, setRowSelection] = React.useState({})
-    const [columnVisibility, setColumnVisibility] = React.useState<VisibilityState>({})
-    const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>([])
-    const [sorting, setSorting] = React.useState<SortingState>([])
+    const [rowSelection, setRowSelection] = useState({})
+    const [columnVisibility, setColumnVisibility] = useState<VisibilityState>({})
+    const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([])
+    const [sorting, setSorting] = useState<SortingState>([])
+
+    const [pagination, setPagination] = useState<PaginationState>({
+        pageIndex: 1,
+        pageSize: 4,
+    })
+
+    const [pageCount, setPageCount] = useState(0)
+    const totalStudents = useStudents()?.totalStudents ?? 0
```

Comment:
```suggestion
    const { totalStudents =  0 } = useStudents()
```

I believe this does the same thing, but test it. This is a nit because it's cleaner js than what you did but a style preference on my part

Reply:
I get this error:
Property 'totalStudents' does not exist on type 'StudentsContextType | undefined'.ts(2339)

What about this?
```
    const { totalStudents = 0 } = useStudents() || {}
```

Reply:
Sure, this works